### PR TITLE
Add support for virtualized InfiniBand testing

### PR DESCRIPTION
**Pull request by frankenmichl**
Configure a defined number of virtual functions for the Mellanox
ConnectX-5 cards we have for InfiniBand testing.

This is a work in progress, what is missing:
 - creating virtual machines with passthrough of virtual functions
 - testing, testing, testing

- Related ticket: https://progress.opensuse.org/issues/63754
- Verification run: pending+

### CODE REVIEW

**Comment by pevik** [comment]
Looks promising :)

**Comment by okurz** [comment]
Congratulations, PR number 10000! Let's celebrate :D

File: tests/kernel/mlx5_sriov.pm [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+    assert_script_run('mount -t nfs openqa.suse.de:/var/lib/openqa/share/factory /mnt/openqa');
+
+    assert_script_run('mkdir -p /var/lib/libvirt/images');
+    assert_script_run("cp $image /var/lib/libvirt/images/");
+
+    # for each desired image do:
+    #   qemu-img create -f qcow2 \
+    #            -o backing_file=/var/lib/libvirt/$image \
+    #            /var/lib/libvirt/$imageNN.qcow2
+
```
Comment (czerw) [inline tests/kernel/mlx5_sriov.pm:89]: qcow images generated by openQA usually wait at grub for keyboard input, do you count with that?

Comment (frankenmichl) [reply tests/kernel/mlx5_sriov.pm:89]: That's actually a good point. No, I did not have this in mind - maybe I need to see if I need an alternative here.

**Comment by frankenmichl** [comment]
> Congratulations, PR number 10000! Let's celebrate :D
And all I got was that lousy shirt?

**Pull request closed without merging**